src/main/java/edu/oregonstate/cs361/battleship/Main.java

```diff
@@ -303,23 +309,28 @@ private static String fireAtNew(Request req) {
         }
 
         j = model.getComputerHits().size();
-        Point Clipper = new Point((PArray[3].getStart().getAcross() + PArray[3].getEnd().getAcross())/2,(PArray[3].getStart().getDown() + PArray[3].getEnd().getDown())/2);
+        //Point Clipper = new Point((PArray[3].getStart().getAcross() + PArray[3].getEnd().getAcross())/2,(PArray[3].getStart().getDown() + PArray[3].getEnd().getDown())/2);
 
 
         // The following branch tree checks if a point fired at
         // BY A PLAYER has hit a COMPUTER ship and adds the point to the array of hits if so
         for (int i = 0; i < 5; i++) {
             if (Hit(CArray[i].getStart(), CArray[i].getEnd(), FireSpot)) {
-                /*if(i == 2){
-                    model.addPointtoArray(PArray[i].getStart(), model.getComputerHits());
-                    model.addPointtoArray(PArray[i].getEnd(), model.getComputerHits());
-                    model.addPointtoArray(Clipper, model.getComputerHits());
 
-                }else {*/
-                    model.addPointtoArray(FireSpot, model.getComputerHits());
-                //}
 
+                model.addPointtoArray(FireSpot, model.getComputerHits());
+                //System.out.println("Player");
+                //System.out.println(model.getShipByID(CArray[i].getName()).getHealth());
                 model.getShipByID(CArray[i].getName()).setHealth(CArray[i].getHealth() - 1);
+                /*System.out.println(model.getShipByID(CArray[i].getName()).getHealth());
+
+                //This is for sinking the ship.
+                if(model.getShipByID(CArray[i].getName()).getHealth() == 0){
+                    model = Sink(model.getShipByID(CArray[i].getName()).getStart(), model.getShipByID(CArray[i].getName()).getEnd(), true, model);
+                    System.out.println("Sink: " + model.getShipByID(CArray[i].getName()).getHealth());
+                }else{
+                    System.out.println("Didn't Sink health: " + model.getShipByID(CArray[i].getName()).getHealth());
+                }*/
```

Comment:
pseudo code for sinking and will either add it or delete it on the final. This is what your issue is about.